cmd/root.go

```diff
@@ -74,15 +80,64 @@ func run(cmd *cobra.Command, args []string) {
 		}
 	}()
 
+	id, err := os.Hostname()
+	if err != nil {
+		log.Fatalf("Unable to get hostname: %v", err)
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	kubeClient := kube.GetInstance()
+	lock, err := resourcelock.New(
+		resourcelock.LeasesResourceLock,
+		namespace,
+		"astro",
+		kubeClient.Client.CoreV1(),
+		kubeClient.Client.CoordinationV1(),
+		resourcelock.ResourceLockConfig{
+			Identity: id,
+		},
+	)
+	if err != nil {
+		log.Fatalf("Unable to create leader election lock: %v", err)
+	}
+
+	leaderelection.RunOrDie(ctx, leaderelection.LeaderElectionConfig{
+		Lock:          lock,
+		LeaseDuration: defaultLeaseDuration,
+		RenewDeadline: defaultRenewDeadline,
+		RetryPeriod:   defaultRetryPeriod,
+		Callbacks: leaderelection.LeaderCallbacks{
+			OnStartedLeading: func(ctx context.Context) {
+				run(ctx, cancel)
+			},
+			OnStoppedLeading: func() {
+				log.Infof("%s leaving", id)
```

Comment:
I feel like there should be a `ctx.Done()` or something here. Does that happen automatically?

Reply:
The `OnStoppedLeading` function actually fires _after_ the `ctx.Done()` call which is performed in the controller [here](https://github.com/FairwindsOps/astro/pull/64/files#diff-3c862eb54a8e0e161b534e0c67e5379eR160)

Context took the place of our stop channel implementation from before.